### PR TITLE
await closing the controller

### PR DESCRIPTION
**Pull request by reteps**
The code on main should await to properly shutdown, but it does not.

### CODE REVIEW

**Comment by reteps** [comment]
Though I believe this would need to be done properly with capturing that...
https://stackoverflow.com/a/61067369/8761914

File: opendevin/server/agent/agent.py [hunk 1]
```diff
@@ -138,6 +138,10 @@ async def on_event(self, event: Event):
...
         return
 
-    def close(self):
+    async def close(self):
+        """Cleanly exits an AgentUnit.
+
+        Execution awaited by the AgentManager.
+        """
         if self.controller is not None:
-            self.controller.close()
```
Comment (yufansong) [inline opendevin/server/agent/agent.py:143]: This change looks make sense. But I thought if we try to call async function in a normal function, our lint or python interpreter wil throw warning during running. Why we didn't find this problem before? 
For example, when I try this simple demo, the interpreter will throw warning. cc @li-boxuan  Do you have any idea?
image

Comment (yimothysu) [reply opendevin/server/agent/agent.py:143]: @yufansong does calling `hello()` print hello world? I thought the coroutine won't execute unless you await it.

Comment (yufansong) [reply opendevin/server/agent/agent.py:143]: Calling `hello` not print, so the interpreter will report the warning. I just try to reproduce the logic in this case which also not use `await`.

Comment (yufansong) [reply opendevin/server/agent/agent.py:143]: > I'm not sure, but I had seen that warning when I shutdown the application with control-c.

If you see the warning, that make sense!

**Review by rbren — APPROVED** [review]
Thanks for the fix!

**Pull request merged**